core/Dispatch.cpp

```diff
@@ -31,38 +32,96 @@ namespace olympia
+                    in_credit_port->enableCollection(node);
+
+                    auto & out_inst_port = out_inst_ports_.emplace_back
+                        (new sparta::DataOutPort<InstQueue::value_type>(&unit_port_set_, "out_"+iq_name+"_write"));
+                    dispatchers_[target_itr->second].emplace_back(new Dispatcher(iq_name,
                                                                              this,
                                                                              info_logger_,
                                                                              in_credit_port.get(),
                                                                              out_inst_port.get()));
+                }
+            }
+        }
+
+        /*
+            We're creating a pipe mapping for each issue queue types:
+            pipe_map_ = {"int": ["issue_queue"]}
+            so we have to find for each pipe, what issue queue is mapped to this, which is a search over
+            pipe_topology and issue_queue_topology
+            For example if we have the topology defined for alus as:
+            pipe_topology_alu_pipes:
+                [["int"], # alu0
+                ["int", "div"], # alu1
+                ["int", "mul"], # alu2
+                ["int", "mul", "i2f", "cmov"], # alu3
+                ["int"], # alu4
+                ["int"] # alu5]
+            and an issue queue topology of:
+            issue_queue_topology:
+                [["alu0"], # issue queue 0
+                ["alu1", "alu2"], # issue queue 1
+                ["alu3", "alu4", "alu5"], # issue queue 2
+                ["fpu0", "fpu1"],
+                ["br0", "br1"]]
+            The algorithm below loops over the issue_queue_topology, finds the associated pipe topology for it,
+            then looks up the associated issue queue dispatcher to associate with that pipe topology.
+
+            Example flow:
+                Iterate onto alu0, look at pipe_toplology_alu_pipes[0] which is "int", 
+                set in pipe_map_ "int": issue_queue0 dispatcher
+        */
+       auto setup_pipe_map = [this, issue_queue_pipe_topology, node](std::string exe_unit){
+            auto exe_unit_pipe_topology = olympia::coreutils::getPipeTopology(node->getParent(), "pipe_topology_" + exe_unit + "_pipes");
+            std::unordered_map<std::string, std::vector<std::string>> exe_unit_to_pipe_mapping;
+            // Get mapping of execution 
+            for(size_t i = 0; i < exe_unit_pipe_topology.size(); ++i){
+                std::string unit_name = exe_unit + std::to_string(i);
+                exe_unit_to_pipe_mapping[unit_name] = exe_unit_pipe_topology[i];
+            }
+            std::unordered_map<std::string, std::shared_ptr<Dispatcher>> string_to_dispatcher;
+            const auto target_itr = InstArchInfo::dispatch_target_map.find(exe_unit);
+            auto & dispatchers = dispatchers_[target_itr->second];
```

Comment:
Missing assert for possible `end()` situation (i.e. not found).  Pls add a GOOD error message for that.

Reply:
Resolved in [6b0e363](https://github.com/riscv-software-src/riscv-perf-model/pull/140/commits/6b0e3633c1b7331170afda8920f596f9fd16b233)